dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/ModuleSelectionConfigImpl.java

```diff
@@ -0,0 +1,459 @@
+                String inOrOut = unmatchedInclude.moduleName != null ? "includeModule" : "includeApplication";
+
+                Tr.warning(tc, MessageConstants.OPENAPI_USING_WRONG_NAME_SOURCE_CWWKO1680W, inOrOut, unmatchedInclude.toString(),
+                           includedNotYetSeenButSeenUnderOldNaming.get(unmatchedInclude));
+            }
+        }
+
+        for (String unmatchedInclude : includedNotYetSeen.stream().map(ModuleName::toString).collect(toList())) {
+            Tr.warning(tc, MessageConstants.OPENAPI_MERGE_UNUSED_INCLUDE_CWWKO1667W, Constants.MERGE_INCLUDE_CONFIG, unmatchedInclude);
+        }
+
+        //Now we repeat the process for excluded applications and modules, however this time we only throw a warning message if we get a match on the deployment descriptor's name
+
+        List<ModuleName> excludedNotYetSeen = new ArrayList<>(config.excluded);
+        for (Iterator<ModuleName> iterator = excludedNotYetSeen.iterator(); iterator.hasNext();) {
+            ModuleName moduleName = iterator.next();
+            for (ModuleInfo moduleInfo : moduleInfos) {
+                if (matches(moduleName, moduleInfo)) {
+                    iterator.remove();
+                    break;
+                }
+            }
+        }
+
+        Map<ModuleName, String> excludedNotYetSeenButSeenUnderOldNaming = new HashMap<>();
+        if (config.serverxmlmode && ProductInfo.getBetaEdition()) {
+            for (Iterator<ModuleName> iterator = excludedNotYetSeen.iterator(); iterator.hasNext();) {
+                ModuleName moduleName = iterator.next();
+                for (ModuleInfo moduleInfo : moduleInfos) {
+                    if (matches(moduleName, moduleInfo, false)) {
+                        excludedNotYetSeenButSeenUnderOldNaming.put(moduleName,
+                                                                    moduleInfo.getApplicationInfo().getDeploymentName());
+                        iterator.remove();
+                        break;
+                    }
+                }
+            }
+
+            for (ModuleName unmatchedExclude : excludedNotYetSeenButSeenUnderOldNaming.keySet()) {
+
+                String inOrOut = unmatchedExclude.moduleName != null ? "excludeModule" : "excludeApplication";
+
+                Tr.warning(tc, MessageConstants.OPENAPI_USING_WRONG_NAME_SOURCE_CWWKO1680W, inOrOut, unmatchedExclude.toString(),
+                           excludedNotYetSeenButSeenUnderOldNaming.get(unmatchedExclude));
+            }
+        }
+
+    }
+
+    /**
+     *
+     * @param name a name we're configured to include in the openAPI documentation
+     * @param module an actual module deployed on the server
+     */
+    private boolean matches(ModuleName name, ModuleInfo module) {
+        return matches(name, module, getConfigValues().serverxmlmode);
+    }
+
+    private static boolean matches(ModuleName name, ModuleInfo module, boolean overridenServerXMLMode) {
```

Comment:
Replace `boolean overriddenServerXMLMode` with `MatchingMode matchingMode` where `MatchingMode` is

```java
private enum MatchingMode {
    SERVER_XML_NAME,
    DEPLOYMENT_DESCRIPTOR_NAME
}
```